GVFS/GVFS.Platform.Mac/MacProductUpgraderPlatformStrategy.cs

```diff
@@ -0,0 +1,63 @@
+using GVFS.Common;
+using GVFS.Common.FileSystem;
+using GVFS.Common.Tracing;
+using System;
+using System.IO;
+
+namespace GVFS.Platform.Mac
+{
+    public class MacProductUpgraderPlatformStrategy : ProductUpgraderPlatformStrategy
+    {
+        public MacProductUpgraderPlatformStrategy(PhysicalFileSystem fileSystem, ITracer tracer)
+        : base(fileSystem, tracer)
+        {
+        }
+
+        public override bool TryPrepareLogDirectory(out string error)
+        {
+            error = null;
+            return true;
+        }
+
+        public override bool TryPrepareApplicationDirectory(out string error)
+        {
+            string rootDirectoryPath = ProductUpgraderInfo.GetUpgradesDirectoryPath();
+            string toolsDirectoryPath = Path.Combine(rootDirectoryPath, ProductUpgrader.ToolsDirectory);
+
+            Exception deleteDirectoryException;
+            if (this.FileSystem.DirectoryExists(toolsDirectoryPath) &&
+                !this.FileSystem.TryDeleteDirectory(toolsDirectoryPath, out deleteDirectoryException))
+            {
+                error = $"Failed to delete {toolsDirectoryPath} - {deleteDirectoryException.Message}";
+
+                this.TraceException(deleteDirectoryException, nameof(this.TryPrepareApplicationDirectory), $"Error deleting {toolsDirectoryPath}.");
+                return false;
+            }
+
+            this.FileSystem.CreateDirectory(toolsDirectoryPath);

```

Comment:
Upgrade verb runs elevated on the Mac too, right? Would this end up creating a root:admin owned directory inside user's `$HOME/Application\ Support/GVFS`? (root owned directory inside user home will be unusual.)

Reply:
> Would this end up creating a root:admin owned directory inside user's $HOME/Application\ Support/GVFS?

Good question - the "application" directory will be created under /usr/local/vfsforgit_upgrader, which is root owned.